events/janus_rabbitmqevh.c

```diff
@@ -0,0 +1,724 @@
+				 * information about the health of a media session, or about
+				 * what's going on in general (e.g., when Janus started/stopped
+				 * receiving media of a certain type, or (TODO) when some media related
+				 * statistics are available). Here's an example of Janus getting
+				 * video from the peer for the first time, or after a second
+				 * of no video at all (which would have triggered a "receiving": false):
+					{
+					   "type": 32,
+					   "timestamp": 3571078797,
+					   "session_id": 2004798115,
+					   "handle_id": 3708519405,
+					   "event": {
+						  "media": "video",
+						  "receiving": "true"
+					   }
+					}
+				*/
+				break;
+			case JANUS_EVENT_TYPE_PLUGIN:
+				/* This is a plugin related event. Since each plugin may
+				 * provide info in a very custom way, the format of this event
+				 * is in general very dynamic. You'll always find, though,
+				 * an "event" object containing the package name of the
+				 * plugin (e.g., "janus.plugin.echotest") and a "data"
+				 * object that contains whatever the plugin decided to
+				 * notify you about, that will always vary from plugin to
+				 * plugin. Besides, notice that "session_id" and "handle_id"
+				 * may or may not be present: when they are, you'll know
+				 * the event has been triggered within the context of a
+				 * specific handle session with the plugin; when they're
+				 * not, the plugin sent an event out of context of a
+				 * specific session it is handling. Here's an example:
+					{
+					   "type": 64,
+					   "timestamp": 3570336031,
+					   "session_id": 2004798115,
+					   "handle_id": 3708519405,
+					   "event": {
+						  "plugin": "janus.plugin.echotest",
+						  "data": {
+							 "audio_active": "true",
+							 "video_active": "true",
+							 "bitrate": 0
+						  }
+					   }
+					}
+				*/
+				break;
+			case JANUS_EVENT_TYPE_TRANSPORT:
+				/* This is a transport related event (TODO). The syntax of
+				 * the common format (transport specific data aside) is
+				 * exactly the same as that of the plugin related events
+				 * above, with a "transport" property instead of "plugin"
+				 * to contain the transport package name. */
+				break;
+			case JANUS_EVENT_TYPE_CORE:
+				/* This is a core related event. This can contain different
+				 * information about the health of the Janus instance, or
+				 * more generically on some events in the Janus life cycle
+				 * (e.g., when it's just been started or when a shutdown
+				 * has been requested). Considering the heterogeneous nature
+				 * of the information being reported, the content is always
+				 * a JSON object (event). Core events are the only ones
+				 * missing a session_id. Here's an example:
+					{
+					   "type": 256,
+					   "timestamp": 28381185382,
+					   "event": {
+						  "status": "started"
+					   }
+					}
+				*/
+				break;
+			default:
+				JANUS_LOG(LOG_WARN, "Unknown type of event '%d'\n", type);
+				break;
+		}
+
```

Comment:
No grouping management as done in the Sample event handler? It might greatly reduce the protocol overhead, when the queue contains more than one pocket as you read from it. The same code we have in the other handler should be re-usable exactly as it is.